src/utils.ts

```diff
@@ -436,6 +436,35 @@ export function* makeCounter(seed = 0): Generator<number> {
   }
 }
 
+/**
+ * Helper for handling legacy callback support.
+ */
+export function maybeCallback<T>(
+  promiseFn: () => Promise<T>,
+  callback?: Callback<T>
+): Promise<T> | void {
+  const PromiseConstructor = PromiseProvider.get();
+
+  const promise = promiseFn();
+  if (callback == null && PromiseConstructor == null) {
+    return promise;
+  }
+
+  if (PromiseConstructor != null) {
```

Comment:
We also need to check here that callback is not defined - otherwise, if a user defines a custom promise library, they can never use callbacks

Reply:
Fixed!